packages/peregrine/lib/talons/App/useApp.js

```diff
@@ -40,6 +46,14 @@ export const useApp = props => {
     } = props;
 
     const [isHTMLUpdateAvailable, setHTMLUpdateAvailable] = useState(false);
+    const [fetchCartId] = useMutation(CREATE_CART_MUTATION);
+    const [, { createCart }] = useCartContext();
+
+    useEffect(() => {
+        createCart({
+            fetchCartId
+        });
+    }, [createCart, fetchCartId]);
```

Comment:
If we like the idea of fetching details within the context provider we may want to consider doing this there as well since it just makes that whole slice cleaner.